source/kraken/realtime.cpp

```diff
@@ -635,6 +635,10 @@ void handle_realtime(const std::string& id,
             meta_vj = data.pt_data->meta_vjs.emplace(trip_update.trip().trip_id());
             // TODO : pick a meaningful TZ
             meta_vj->tz_handler = data.pt_data->tz_manager.get_first_timezone();
+        } else if (is_cancelled_trip(trip_update)) {
+            LOG4CPLUS_DEBUG(log, "Can't cancelled a trip that doesn't exist: ignoring trip update id "
+                    << trip_update.trip().trip_id());
```

Comment:
Don't we want to return some kind of error in the protobuf ? What would be the end result from kirin in this case ?

Reply:
The error managing is not complete. 
In many case, we have only a log message to trace the error concerned.
The main job is to protect the code against `segfault or bad behaviours` inside Kraken engine.
We don't reject the flow in Kirin because the *COTS* format is good, only write a `Can't cancelled a trip that doesn't exist: ignoring trip update id` log Kraken message.

A good upgrade would be to fill *Prometheus probe* to track the behaviour. I have looked and is not easy to plug it at this code level. That needs a lot of changes. Maybe in the near future

Reply:
I agree with @benoit-bst .

There would be a lot of levels involved in such a process: kirin doesn't listen anything from kraken, it only publishes updates in a rabbitMQ queue. And Kirin checks if the trip exists before sending it to kraken.
But then again, I'm pretty sure that no-one reads kirin's responses to COTS feed POST... And the tool posting COTS is just reading a queue. So no specific point in responding to the feed received.

Reply:
:+1: